### PR TITLE
Convert and package

### PR DESCRIPTION
**Pull request by laurenssam**
Pull requests contains:

Converted the triangulation to work with COCO format files:
- Triangulation determines the centre point of the object based on the segmentation map
- Takes as input as coco JSON file and output a csv file with lonlat coordinates
- Minimized the code to only what we need for now

Packaged the triangulation algorithm:
- Please check if the installation works for you!

### CODE REVIEW

File: triangulation/helpers.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            location = models.LocationQuery(
+                latitude=float(row[0]), longitude=float(row[1]), radius=100
+            )
+            query_result: models.PagedPanoramasResponse = PanoramaClient.list_panoramas(
+                location=location,
+                timestamp_after=timestamp_after,
+                timestamp_before=timestamp_before)
+
+            # Download the corresponding image to your machine
+            PanoramaClient.download_image(query_result.panoramas[0], output_location=output_folder,
```
Comment (epureanudiana) [inline triangulation/helpers.py]: Is the first element in the `query_result` list the closest one? Maybe the radius could be set to some smaller value just to be safe.

File: triangulation/helpers.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    transform = osr.CoordinateTransformation(source, target)
+    point.Transform(transform)
+
+    return [point.GetX(), point.GetY()]
+
+
+def get_panos_from_points_of_interest(point_of_interest, output_folder, timestamp_before, timestamp_after):
+    """
+    Downloads the panoramas that are closest to the points of interest
```
Comment (epureanudiana) [inline triangulation/helpers.py]: I could to use this function as well when I am creating the map, but I do not want to download the panoramas. 
I suggest we change the return type of this function to List[Panorama] and download the panoramas separately.

```
def get_panos_from_points_of_interest(point_of_interest, output_folder, timestamp_before, timestamp_after):

     """
     Retrieves the panoramas that are closest to the points of interest
     Takes as input a csv file with lon, lat coordinates outputs images in the output folder
     """

     if isinstance(output_folder, str):
         output_folder = Path(output_folder)
     output_folder.mkdir(exist_ok=True, parents=True)

     panoramas = []
     with open(point_of_interest, 'r') as file:
         reader = csv.reader(file)
         next(reader) #skip first line
         for row in reader:
             location = models.LocationQuery(
                 latitude=float(row[0]), longitude=float(row[1]), radius=100
             )
             query_result: models.PagedPanoramasResponse = PanoramaClient.list_panoramas(
                 location=location,
                 timestamp_after=timestamp_after,
                 timestamp_before=timestamp_before)
             panoramas.append(query_result[0])
```

And then 
```
panoramas = get_panos_from_points_of_interest(...)

 # Download the corresponding image to your machine 
for panorama in panoramas:
    PanoramaClient.download_image(panorama, output_location=output_folder, size=models.ImageSize.MEDIUM)
```

Alternatively, have a function to download a list of panorama objects.

File: triangulation/helpers.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+def euclidean_distance(x_1, y_1, x_2, y_2):
+    """
+    Calculate the Euclidean distance between two vectors
+    """
+    return sqrt(((x_1 - x_2)**2) + ((y_1 - y_2)**2))
+
+def get_pano_location(pano_id):
+    """
+    Get the initial location coordinates of a panoramic image
+    and convert it from WGS84 (EPSG:4326) to Rijksdriehoek (EPSG:28992)
```
Comment (epureanudiana) [inline triangulation/helpers.py:49]: When I plot the daily trajectory of the car I am using the lat, long from the API. 

Do I need to convert them as well to Rijksdriehoek (EPSG:28992)?

**Review by epureanudiana — COMMENTED** [review]
Nice!
Can you write a short explanation of what is happening with the 2 coordinate systems, i.e.
WGS84 (EPSG:4326) Rijksdriehoek (EPSG:28992). Or just let me know if my understanding is correct: 
- osgeo, thus the triangulation algorithm works with RD
- panorama API works with WGS84
- I do not need to worry about these conversion in the maps that I make.

File: triangulation/helpers.py (hunk 3 above)
Comment (laurenssam) [reply triangulation/helpers.py:49]: No, I don't know why, but the original code is working with the Rijksdriekhoek coordinate system. We can just long, lat which is more generic

**Pull request merged**